### PR TITLE
KHS-62 Fixed Thing-State ManyToMany

### PR DESCRIPTION
**Pull request by dmarinen**

### CODE REVIEW

**Review by ExiledRain — APPROVED** [review]

**Review by ExiledRain — APPROVED** [review]

File: src/main/java/com/knits/coreplatform/web/rest/DeviceResource.java [hunk 1]
```diff
@@ -110,7 +106,6 @@ public DeviceResource(DeviceService deviceService, DeviceRepository deviceReposi
      * @throws URISyntaxException if the Location URI syntax is incorrect.
      */
     @PatchMapping(value = "/devices/{id}", consumes = "application/merge-patch+json")
-    @PreAuthorize("hasAuthority(\"" + AuthoritiesConstants.COMPANY_ADMIN + "\")")
```
Comment (ExiledRain) [inline src/main/java/com/knits/coreplatform/web/rest/DeviceResource.java]: Is it ok that you lose all the changes?

File: src/main/java/com/knits/coreplatform/web/rest/DeviceResource.java [hunk 2]
```diff
@@ -141,7 +136,6 @@ public DeviceResource(DeviceService deviceService, DeviceRepository deviceReposi
      * @return the {@link ResponseEntity} with status {@code 200 (OK)} and the list of devices in body.
      */
     @GetMapping("/devices")
-    @PreAuthorize("hasAnyAuthority(\"" + AuthoritiesConstants.COMPANY_ADMIN + "\",\"" + AuthoritiesConstants.COMPANY_ADMIN + "\")")
```
Comment (ExiledRain) [inline src/main/java/com/knits/coreplatform/web/rest/DeviceResource.java]: Another lost update

File: src/main/java/com/knits/coreplatform/web/rest/DeviceResource.java [hunk 3]
```diff
@@ -66,15 +63,14 @@ public DeviceResource(DeviceService deviceService, DeviceRepository deviceReposi
...
-     * @param id        the id of the deviceDTO to save.
+     * @param id the id of the deviceDTO to save.
      * @param deviceDTO the deviceDTO to update.
      * @return the {@link ResponseEntity} with status {@code 200 (OK)} and with body the updated deviceDTO,
      * or with status {@code 400 (Bad Request)} if the deviceDTO is not valid,
      * or with status {@code 500 (Internal Server Error)} if the deviceDTO couldn't be updated.
      * @throws URISyntaxException if the Location URI syntax is incorrect.
      */
     @PutMapping("/devices/{id}")
-    @PreAuthorize("hasAuthority(\"" + AuthoritiesConstants.COMPANY_ADMIN + "\")")
```
Comment (ExiledRain) [inline src/main/java/com/knits/coreplatform/web/rest/DeviceResource.java]: Lost authority

File: src/main/java/com/knits/coreplatform/web/rest/DeviceResource.java [hunk 4]
```diff
@@ -154,7 +148,6 @@ public DeviceResource(DeviceService deviceService, DeviceRepository deviceReposi
      * @return the {@link ResponseEntity} with status {@code 200 (OK)} and with body the deviceDTO, or with status {@code 404 (Not Found)}.
      */
     @GetMapping("/devices/{id}")
-    @PreAuthorize("hasAnyAuthority(\"" + AuthoritiesConstants.COMPANY_ADMIN + "\",\"" + AuthoritiesConstants.COMPANY_ADMIN + "\")")
```
Comment (ExiledRain) [inline src/main/java/com/knits/coreplatform/web/rest/DeviceResource.java]: Lost authority

File: src/main/java/com/knits/coreplatform/web/rest/LocationResource.java [hunk 5]
```diff
@@ -50,7 +48,6 @@ public LocationResource(LocationService locationService, LocationRepository loca
      * @throws URISyntaxException if the Location URI syntax is incorrect.
      */
     @PostMapping("/locations")
-    @PreAuthorize("hasAuthority(\"" + AuthoritiesConstants.COMPANY_ADMIN + "\")")
```
Comment (ExiledRain) [inline src/main/java/com/knits/coreplatform/web/rest/LocationResource.java]: Lost authority

**Review by ExiledRain — CHANGES_REQUESTED** [review]

**Comment by dmarinen** [comment]
Just checked the code directly on GitHub and mentioned  problems are not there.

**Pull request closed without merging**